CHANGELOG.md

```diff
@@ -32,6 +32,7 @@ We use *breaking* word for marking changes that are not backward compatible (rel
 
 ### Added
 
+- [#3031](https://github.com/thanos-io/thanos/pull/3031) Compact/Sidecar/other writers: added `--add-hashes` and `--shipper.add-hashes` respectively. If enabled, they calculate SHA256 hashes of each file in a block before uploading them. If those hashes exist in the `meta.json` file then Compact becomes smart and does not download the files if they already exist on disk. This also means that the data directory passed to Thanos Compact is only *cleared once at boot*. So, if you, for example, use persistent volumes on k8s and your Thanos Compact crashes or fails to make an iteration properly then the last downloaded files are not wiped from the disk. The directories are only wiped again after a successful iteration.
```

Comment:
Does it infinitely save data on disk and never delete it? :thinking: This is what I read from this

Reply:
>  So, if you, for example, use persistent volumes on k8s and your Thanos Compact crashes or fails to make an iteration properly then the last downloaded files are not wiped from the disk

From the code it looks like they ARE wiped in fact, no?

Reply:
Ok I think we need to improve this (really 2x reduce) description, and just say "on-crash & recovery we don't delete data anymore but we are able to start up without redownload.